### ISSUE TITLE
can you show your training log?

### ISSUE THREAD

**Issue by Allan9977** [meownoid/face-identification-tpe#5]
Hi,I've already run the demo. I want to train the tpe(the fc layer) by mine samples(the cnn feature extracted by caffe),but the loss is hard to reduce,and the dev err almost the same,can you show your training log? thanks!

**Comment by meownoid** [comment]
Something like this:
```
Using TensorFlow backend.
epoch: 0
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.4356      
EER: 20.47
epoch: 1
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.4117     
EER: 18.32
epoch: 2
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3957     
EER: 16.16
epoch: 3
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.4083     
EER: 15.63
epoch: 4
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.4040     
EER: 15.97
epoch: 5
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3995     
EER: 15.55
epoch: 6
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3859     
EER: 14.65
epoch: 7
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3949     
EER: 15.15
epoch: 8
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3881     
EER: 13.74
epoch: 9
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3850     
EER: 13.85
epoch: 10
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3843     
EER: 14.44

...

epoch: 50
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3689     
EER: 14.87
epoch: 51
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3749     
EER: 15.08
epoch: 52
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3818     
EER: 14.52
epoch: 53
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3766     
EER: 14.15
epoch: 54
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3693     
EER: 14.36
epoch: 55
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3748     
EER: 14.30
epoch: 56
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3789     
EER: 14.32
epoch: 57
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3612     
EER: 13.14
epoch: 58
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3714     
EER: 13.19
epoch: 59
Epoch 1/1
512/512 [==============================] - 0s - loss: 0.3781     
EER: 12.85
```
First thing you should try is to change the optimizer. (`tpe.py`). Try SGD with small learning rate. Try Adam. Try to set `COLD_START = 50` for mining hard-negatives.

**Comment by zhuzzjlu** [comment]
Hi, I would like to know what the dev_protocol is and how to generate it.

**Comment by aalesar** [comment]
Hi , for some reason my loss decreases very well (-from 0.4 to 0.13) , but EER increases (from 0.2 to 0.4) especially after hard mining starts. I can't find a reason for this strange behavior. Can you think of something that goes wrong? Thanks in advance!
